File: app/src/main/java/com/robotnec/budget/core/service/CurrencyExchangeService.java
package com.robotnec.budget.core.service;

import com.robotnec.budget.core.domain.Currency;
import com.robotnec.budget.core.domain.MoneyAmount;

/**
 * @author zak <zak@swingpulse.com>
 */
public interface CurrencyExchangeService {
    MoneyAmount exchange(MoneyAmount from, Currency to);
}

File: app/src/main/java/com/robotnec/budget/core/service/CurrencyExchangeService.kt
package com.robotnec.budget.core.service

import org.joda.money.CurrencyUnit
import org.joda.money.Money

/**
 * @author zak zak@swingpulse.com>
 */
interface CurrencyExchangeService {
    fun exchange(from: Money, to: CurrencyUnit): Money
}
